integration/simulation/validate_chain.go

```diff
@@ -61,7 +61,7 @@ func checkObscuroBlockchainValidity(t *testing.T, s *Simulation, maxL1Height uin
 	}
 
 	min, max := minMax(heights)
-	if max-min > max/10 {
+	if max-min > max/3 {
```

Comment:
Had to reduce it because the test fails with such high precision.

I can't say for sure the source of the issue without more investigation, but my gut feeling is that's related to how we're broadcasting blocks. 
I did check the pattern of values in `heights`, before and after the txinjector change.
They are the same: the first nodes have a smaller height than the last nodes in the array.

These changes produce less total number of txs being injected, so I think it's expected to see a wider gap in the `heights` values.

Reply:
I threw in a TODO. But should we investigate why the heights array behaves the way it does ?

Reply:
I'm still confused here. Even if less txs are being injected, shouldn't there be a standard number of blocks, which is what is being checked here? Or are we skipping empty blocks perhaps?

Reply:
I would expect the roughly same number of blocks too. 

Given that block production is continuous and controlled here: https://github.com/obscuronet/obscuro-playground/blob/d102008ed01aba3461713ea5a8c5c20a68ebaf23/integration/ethereummock/node.go#L236-L240

Unless there's something triggering more often:

https://github.com/obscuronet/obscuro-playground/blob/d102008ed01aba3461713ea5a8c5c20a68ebaf23/integration/ethereummock/node.go#L224-L230

I'm capturing this in a task for investigation, I think it's ok to defer it for now because it's mainly a test around ethereummock block production. 

Maybe @tudor-malene can prioritize the investigation ?